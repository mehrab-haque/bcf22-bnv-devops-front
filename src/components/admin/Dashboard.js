import * as React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';
import Divider from '@mui/material/Divider';
import Drawer from '@mui/material/Drawer';
import IconButton from '@mui/material/IconButton';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import MenuIcon from '@mui/icons-material/Menu';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import {useState} from "react";
import ForumIcon from '@mui/icons-material/Forum';
import AddAlertIcon from '@mui/icons-material/AddAlert';
import LocationCityIcon from '@mui/icons-material/LocationCity';
import DeliveryDiningIcon from '@mui/icons-material/DeliveryDining';
import AssessmentIcon from '@mui/icons-material/Assessment';
import PeopleAltIcon from '@mui/icons-material/PeopleAlt';
import ExitToAppIcon from '@mui/icons-material/ExitToApp';
import {Button, CircularProgress, Dialog, DialogActions, DialogContent, DialogTitle} from "@mui/material";
import Courses from "./Courses";

const drawerWidth = 240;

var showCart

const Dashboard=props=>{
    const { window } = props;
    const [mobileOpen, setMobileOpen] = React.useState(false);

    const handleDrawerToggle = () => {
        setMobileOpen(!mobileOpen);
    };

    const [menu,setMenu]=useState(0)

    const drawer = (
        <div>
            <Toolbar />

            <List>
                <Divider />
                <ListItem button selected={menu===1} onClick={()=>{setMenu(1)}}>
                    <ListItemIcon>
                        <ForumIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Courses'} />
                </ListItem>
                <ListItem button selected={menu===2} onClick={()=>{setMenu(2)}}>
                    <ListItemIcon>
                        <LocationCityIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Students'} />
                </ListItem>
                <Divider />
                <ListItem button selected={menu===3} onClick={()=>{setMenu(3)}}>
                    <ListItemIcon>
                        <AddAlertIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Courses Assignment'} />
                </ListItem>
                <Divider />
                <ListItem button selected={menu===4} onClick={()=>{setMenu(4)}}>
                    <ListItemIcon>
                        <DeliveryDiningIcon/>
                    </ListItemIcon>
                    <ListItemText primary={'Result'} />
                </ListItem>
                <Divider />
            </List>
        </div>
    );

    const container = window !== undefined ? () => window().document.body : undefined;


    return (
        <Box sx={{ display: 'flex' }}>
            <CssBaseline />
            <AppBar
                position="fixed"
                sx={{
                    width: { sm: `calc(100% - ${drawerWidth}px)` },
                    ml: { sm: `${drawerWidth}px` },
                }}
                style={{backgroundColor:'#666666'}}
            >
                <Toolbar>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        edge="start"
                        onClick={handleDrawerToggle}
                        sx={{ mr: 2, display: { sm: 'none' } }}
                    >
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6" noWrap component="div">
                        Admin Dashboard
                    </Typography>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        edge="end"
                        sx={{marginLeft:'auto'}}
                        onClick={props.logout}
                    >
                        <ExitToAppIcon/>
                    </IconButton>
                </Toolbar>
            </AppBar>
            <Box
                component="nav"
                sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
                aria-label="mailbox folders"
            >
                {/* The implementation can be swapped with js to avoid SEO duplication of links. */}
                <Drawer
                    container={container}
                    variant="temporary"
                    open={mobileOpen}
                    onClose={handleDrawerToggle}
                    ModalProps={{
                        keepMounted: true, // Better open performance on mobile.
                    }}
                    sx={{
                        display: { xs: 'block', sm: 'none' },
                        '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth },
                    }}
                >
                    {drawer}
                </Drawer>
                <Drawer
                    variant="permanent"
                    sx={{
                        display: { xs: 'none', sm: 'block' },
                        '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth },
                    }}
                    open
                >
                    {drawer}
                </Drawer>
            </Box>
            <Box
                component="main"
                sx={{ flexGrow: 1, p: 3, width: { sm: `calc(100% - ${drawerWidth}px)` } }}
            >
                <Toolbar />


                {
                    menu===1?(
                        <Courses/>
                    ):(
                        <div/>
                    )
                }




            </Box>
        </Box>
    );
}

Dashboard.propTypes = {
    /**
     * Injected by the documentation to work in an iframe.
     * You won't need it on your project.
     */
    window: PropTypes.func,
};

export default Dashboard
export {showCart}