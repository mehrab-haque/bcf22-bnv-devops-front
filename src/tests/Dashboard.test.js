import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Dashboard from "../components/admin/Dashboard";


describe("Component Dashboard Render", () => {
    it("renders without crashing given the required props", () => {
        const wrapper = shallow(<Dashboard/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
