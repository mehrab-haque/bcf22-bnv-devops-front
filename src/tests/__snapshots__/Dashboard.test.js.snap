// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Dashboard Render renders without crashing given the required props 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "display": "flex",
    }
  }
>
  <CssBaseline />
  <ForwardRef(AppBar)
    position="fixed"
    style={
      Object {
        "backgroundColor": "#666666",
      }
    }
    sx={
      Object {
        "ml": Object {
          "sm": "240px",
        },
        "width": Object {
          "sm": "calc(100% - 240px)",
        },
      }
    }
  >
    <ForwardRef(Toolbar)>
      <ForwardRef(IconButton)
        aria-label="open drawer"
        color="inherit"
        edge="start"
        onClick={[Function]}
        sx={
          Object {
            "display": Object {
              "sm": "none",
            },
            "mr": 2,
          }
        }
      >
        <Memo(ForwardRef(MenuIcon)) />
      </ForwardRef(IconButton)>
      <ForwardRef(Typography)
        component="div"
        noWrap={true}
        variant="h6"
      >
        Admin Dashboard
      </ForwardRef(Typography)>
      <ForwardRef(IconButton)
        aria-label="open drawer"
        color="inherit"
        edge="end"
        sx={
          Object {
            "marginLeft": "auto",
          }
        }
      >
        <Memo(ForwardRef(ExitToAppIcon)) />
      </ForwardRef(IconButton)>
    </ForwardRef(Toolbar)>
  </ForwardRef(AppBar)>
  <ForwardRef(Box)
    aria-label="mailbox folders"
    component="nav"
    sx={
      Object {
        "flexShrink": Object {
          "sm": 0,
        },
        "width": Object {
          "sm": 240,
        },
      }
    }
  >
    <ForwardRef(Drawer)
      ModalProps={
        Object {
          "keepMounted": true,
        }
      }
      onClose={[Function]}
      open={false}
      sx={
        Object {
          "& .MuiDrawer-paper": Object {
            "boxSizing": "border-box",
            "width": 240,
          },
          "display": Object {
            "sm": "none",
            "xs": "block",
          },
        }
      }
      variant="temporary"
    >
      <div>
        <ForwardRef(Toolbar) />
        <ForwardRef(List)>
          <ForwardRef(Divider) />
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(ForumIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Courses"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(LocationCityIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Students"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(Divider) />
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(AddAlertIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Courses Assignment"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(Divider) />
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(DeliveryDiningIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Result"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(Divider) />
        </ForwardRef(List)>
      </div>
    </ForwardRef(Drawer)>
    <ForwardRef(Drawer)
      open={true}
      sx={
        Object {
          "& .MuiDrawer-paper": Object {
            "boxSizing": "border-box",
            "width": 240,
          },
          "display": Object {
            "sm": "block",
            "xs": "none",
          },
        }
      }
      variant="permanent"
    >
      <div>
        <ForwardRef(Toolbar) />
        <ForwardRef(List)>
          <ForwardRef(Divider) />
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(ForumIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Courses"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(LocationCityIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Students"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(Divider) />
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(AddAlertIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Courses Assignment"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(Divider) />
          <ForwardRef(ListItem)
            button={true}
            onClick={[Function]}
            selected={false}
          >
            <ForwardRef(ListItemIcon)>
              <Memo(ForwardRef(DeliveryDiningIcon)) />
            </ForwardRef(ListItemIcon)>
            <ForwardRef(ListItemText)
              primary="Result"
            />
          </ForwardRef(ListItem)>
          <ForwardRef(Divider) />
        </ForwardRef(List)>
      </div>
    </ForwardRef(Drawer)>
  </ForwardRef(Box)>
  <ForwardRef(Box)
    component="main"
    sx={
      Object {
        "flexGrow": 1,
        "p": 3,
        "width": Object {
          "sm": "calc(100% - 240px)",
        },
      }
    }
  >
    <ForwardRef(Toolbar) />
    <div />
  </ForwardRef(Box)>
</ForwardRef(Box)>
`;
